xnmt/evaluator.py

```diff
@@ -536,52 +541,33 @@ def evaluate(self, ref, hyp):
     proc = subprocess.Popen([self.path], stdout=subprocess.PIPE, shell=True)
     (out, _) = proc.communicate()
     external_score = float(out)
-    return ExternalScore(external_score, self.higher_better, desc=self.desc)
+    return ExternalScore(external_score, self.higher_better, desc=desc)
 
-class RecallEvaluator(object):
-  def __init__(self, nbest=5, desc=None):
+class RecallEvaluator(Evaluator,Serializable):
+  yaml_tag = "!RecallEvaluator"
+  @serializable_init
+  def __init__(self, nbest=5):
     self.nbest = nbest
-    self.desc = desc
 
   def metric_name(self):
     return "Recall{}".format(str(self.nbest))
 
-  def evaluate(self, ref, hyp):
+  def evaluate(self, ref, hyp, desc=None):
     true_positive = 0
     for hyp_i, ref_i in zip(hyp, ref):
       if any(ref_i == idx for idx, _ in hyp_i[:self.nbest]):
         true_positive += 1
     score = true_positive / float(len(ref))
-    return RecallScore(score, len(hyp), len(ref), nbest=self.nbest, desc=self.desc)
+    return RecallScore(score, len(hyp), len(ref), nbest=self.nbest, desc=desc)
 
-class MeanAvgPrecisionEvaluator(object):
```

Comment:
Why was this deleted? If it's incompatible, let's just comment it out for now and restore it later. It's necessary for evaluating retrieval models.